### PR TITLE
Make Discussion components dark mode compatible

### LINKED ISSUE #10553: Ensure discussion components are compatible with dark mode

### PR DESCRIPTION
**Pull request by cemms1**
## What does this change?

Ensures that Discussion components use colours from the themed palette definition rather than directly from source.

Also addresses some obvious inconsistencies in the palette colours for dark mode.

There will be follow-up PRs after this one to improve the dark mode designs for discussion components.

## Why?

Allows discussion components to be compatible with dark mode ahead of serving this feature in apps

Resolves #10553

### CODE REVIEW

File: dotcom-rendering/src/components/AdSlot.apps.tsx [hunk 1]
```diff
@@ -99,8 +98,6 @@ const SupportBanner = ({
  * These slots are dynamically inserted into articles using React portals.
  * The ref is important so that we can provide the location of the slot to
  * the native layer, for it to "paint" an advert over the top of it.
- *
- * @todo Style for dark mode
```
Comment (abeddow91) [inline dotcom-rendering/src/components/AdSlot.apps.tsx:103]: Nit and non-blocking: Are there tickets for these as well as comments? Todos can get a bit lost after the pr is merged

File: dotcom-rendering/src/palette.ts [hunk 2]
```diff
@@ -4871,6 +4896,8 @@ const interactiveAtomBackgroundDark: PaletteFunction = () =>
 /**
  * A template literal type used to make sure the keys of the palette use the
  * correct CSS custom property syntax.
+ *
+ * IMPORTANT - do use a CSS variable name ending in "label" as Storybook theming will fail
```
Comment (abeddow91) [inline dotcom-rendering/src/palette.ts:4900]: should this be `do not use` ?

```suggestion
 * IMPORTANT - do not use a CSS variable name ending in "label" as Storybook theming will fail
```

**Review by abeddow91 — APPROVED** [review]
This looks great, thanks for tidying up as you went 🧹

File: dotcom-rendering/src/components/Discussion/Badges.tsx [hunk 3]
```diff
@@ -6,6 +6,8 @@ import {
 } from '@guardian/source-foundations';
 import { palette as themePalette } from '../../palette';
 
+const white = sourcePalette.neutral[100];
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion/Badges.tsx:9]: Is this still correct in dark mode?

**Pull request merged**